src/contract.jl

```diff
@@ -36,3 +40,66 @@ function contract_density_matrix(
   end
   return out
 end
+
+function ITensors.scalar(
+  alg::Algorithm, tn::Union{AbstractITensorNetwork,Vector{ITensor}}; kwargs...
+)
+  return contract(alg, tn; kwargs...)[]
+end
+
+function ITensors.scalar(
+  tn::Union{AbstractITensorNetwork,Vector{ITensor}}; alg="exact", kwargs...
+)
+  return scalar(Algorithm(alg), tn; kwargs...)
+end
+
+function logscalar(
+  tn::Union{AbstractITensorNetwork,Vector{ITensor}}; alg::String="exact", kwargs...
+)
+  return logscalar(Algorithm(alg), tn; kwargs...)
+end
+
+function logscalar(
+  alg::Algorithm"exact", tn::Union{AbstractITensorNetwork,Vector{ITensor}}; kwargs...
+)
+  s = scalar(alg, tn; kwargs...)
+  if s ≈ 0
+    tol = 1e-16
```

Comment:
FYI you need to use the syntax `≈(s, 0; tol=1e-16)` here, `s ≈ 0 tol = 1e-16` only works inside of `@test`.